numpy/lib/tests/test_type_check.py

```diff
@@ -323,8 +323,9 @@ def test_generic(self):
         # perform the same test but in-place
         with np.errstate(divide='ignore', invalid='ignore'):
             vals = np.array((-1., 0, 1))/0.
-        nan_to_num(vals, copy=False)
+        result = nan_to_num(vals, copy=False)
 
+        assert_array_equal(result, vals)
```

Comment:
Heh right. Thanks. I'll squash the commits down once you're comfortable with the review.